ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/StackComponentView.kt

```diff
@@ -171,6 +189,112 @@ private fun StackWithOverlaidBadge(
     }
 }
 
+@Suppress("LongMethod", "LongParameterList", "CyclomaticComplexMethod")
+@Composable
+private fun StackWithEdgeToEdgeBadge(
+    stackState: StackComponentState,
+    state: PaywallState.Loaded.Components,
+    badgeStack: StackComponentStyle,
+    alignment: TwoDimensionalAlignment,
+    clickHandler: suspend (PaywallAction) -> Unit,
+    modifier: Modifier = Modifier,
+    selected: Boolean = false,
+) {
+    var badgeHeight by remember {
+        mutableIntStateOf(0)
+    }
+    val badgeHeightDp = with(LocalDensity.current) { badgeHeight.toDp() }
+    val badge = @Composable {
+        StackComponentView(
+            // The background and border is applied to the parent container so we null it out here.
+            // We make the badge use all the available width without increasing the size of the main content.
+            badgeStack.copy(
+                backgroundColor = null,
+                size = Size(width = Fill, height = badgeStack.size.height),
+                border = null,
+                margin = PaddingValues(0.dp),
```

Comment:
We basically clear the properties that are already applied to the background composable so they are not applied again to the badge itself.